lib/modules/apostrophe-doc-type-manager/lib/api.js

```diff
@@ -31,9 +32,7 @@ module.exports = function(self, options) {
       // So methods called later, or extending this method, see the change in piece
       doc[name] = $set[name];
     });
-    if (_.isEmpty($set)) {
-      return setImmediate(callback);
-    }
+    $set.trash = true;
```

Comment:
The responsibility for setting the `trash` property itself should not be added to this existing method, because existing overrides of the method that currently provide a compatible implementation would fail to move things to the trash (bc break). Instead it needs to be handled as before in `trashInSchema` itself. Needs to get cleared there when we're actually rescuing, too.

Reply:
OK, I guess this is a small performance hit - add'l database access? But not a big deal to change :)